homeassistant/components/tplink/__init__.py

```diff
@@ -136,33 +147,66 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             ", ".join(d.host for d in switches),
         )
 
+    async def async_retry_devices(self) -> None:
+        """Retry unavailable devices."""
+        unavailable_devices: list[SmartDevice] = hass_data[UNAVAILABLE_DEVICES]
+        _LOGGER.debug(
+            "retry during setup unavailable devices: %s",
+            [d.host for d in unavailable_devices],
+        )
+
+        for device in unavailable_devices:
+            try:
+                device.get_sysinfo()
+            except SmartDeviceException:
+                continue
+            _LOGGER.debug(
+                "at least one device is available again, so reload integration"
+            )
+            await hass.config_entries.async_reload(entry.entry_id)
+
     # prepare DataUpdateCoordinators
-    hass.data[DOMAIN][COORDINATORS] = {}
+    hass_data[COORDINATORS] = {}
     for switch in switches:
 
         try:
             await hass.async_add_executor_job(switch.get_sysinfo)
-        except SmartDeviceException as ex:
-            _LOGGER.debug(ex)
-            raise ConfigEntryNotReady from ex
+        except SmartDeviceException:
+            _LOGGER.warning(
+                "Device at '%s' not reachable during setup, will retry later",
+                switch.host,
+            )
+            unavailable_devices.append(switch)
+            continue
 
-        hass.data[DOMAIN][COORDINATORS][
+        hass_data[COORDINATORS][
             switch.mac
         ] = coordinator = SmartPlugDataUpdateCoordinator(hass, switch)
 
         await coordinator.async_config_entry_first_refresh()
 
+    if unavailable_devices:
+        hass_data[UNAVAILABLE_REMOVE_LISTENER] = async_track_time_interval(
+            hass, async_retry_devices, UNAVAILABLE_RETRY_DELAY
+        )
+        unavailable_devices_hosts = [d.host for d in unavailable_devices]
```

Comment:
No need for this, you can store it on the config entry.

`entry.async_on_unload(async_track_time_interval(…))`

Reply:
uhh nice!